cheatsheets_draft/Webhook_Security_Guidelines_CheatSheet.md

```diff
@@ -0,0 +1,169 @@
+
+It’s recommended to include a signature header as part of POST request, this allows additional layer of security to ensure requests are originating from legitimate source. You can have your own solution for this or leverage official libraries from third party integration. 
+
+Abuse Case: 
+
+- Using a targeted webhook capture a message and replay it with a removed or replaced HMAC. If the service accepts the message it is vulnerable to this attack as it is not validating the HMAC
+
+
+## Control 8 - Cross Site Request Forgery 
+
+CSRF protection - CSRF Protection for webhooks is an important Security feature to prevent Cross Site Request Forgery. If you’re using Rails, Django, or another web framework, your site might automatically check that every POST request contains a CSRF token. However, this security measure might also prevent your site from processing legitimate requests. If so, you might need to exempt the webhooks route from CSRF protection and handle them. 
+
+Abuse Case: 
+
+- Using a targeted webhook try sending a webhook POST request with a known list of malformed Cross-Site Request Forgery payloads.
+
+
+## Control 9 - Rotation and Retry 
+
+Retry Logic - Have provision to reset a webhook's authentication key. To ensure that you do not lose any webhook requests between the time you reset your key and when you update your application to start using that new key, your webhook processor should reject requests with failed signatures with custom Status code. 
+
+Abuse Case: 
+
+- Using a targeted webhook, capture HTTP Request with Auth key. Generate new Authkey for the Webhook URL, try POST ing with expired Auth key, Invalid Authkey, payload and validate.
+
+
+## Control 10 - Pass Secrets in Request Header 
+
+Protection from repudiation attacks - Most webhook providers allow you to pass a secret or a "token" in every webhook request. Example Github. The secret is typically passed as an HTTP header, as a field in the JSON payload or appended to the request URL. When you confirm the token you received is the one you expected, it helps validate the request was sent by the webhook publisher. 
+
+Abuse Case: 
+
+- Using a targeted webhook capture a known good message. Tamper or remove the Token from Request Header and POST the message to target URL. The webhook should respond that the message was invalid. If it does not it is vulnerable to repudiation attacks
+
+
+## Control 11 - Fail Safe and Secure 
```

Comment:
The abuse case scenario and the description provided do not relate in any way.
In the description, the business logic is being described in order to receive one-time requests and have somewhat a chain of events that need to be followed and be strict in following it.

In the abuse case, you're simply saying that if someone sends a bogus request, to fail securely.

Reply:
Ack, Thank you @ThunderSon. 

I think Replay and Payload Best practices covers the abuse for this one. 

Replay Attack covers abuse around resending already processed request to the Webhook listener and how validating timestamp can help. Payload validations can help to identify what kind of data is coming to listener. 

I would like to remove this, please advice.

Reply:
No need to remove this. Fail secure is a control that should be mentioned. How could webhooks fail in a bad manner? Maybe execute the code despite failing, or returning certain information that shouldn't be disclosed because they followed a fail safe method or a fail open, just to keep their users happy. This definitely should come after control 12 and 13.